catalogue/webapp/components/WorkDetails/WorkDetails.js

```diff
@@ -256,89 +296,78 @@ const WorkDetails = ({
-            <Space
-              v={{
-                size: 'm',
-                properties: ['margin-top'],
-              }}
-            >
-              <p
+              <div
                 className={classNames({
-                  'no-margin': true,
-                  [font('lr', 6)]: true,
+                  'flex flex-h-center': true,
                 })}
               >
-                Contains:{' '}
-                {childManifestsCount > 0
-                  ? `${childManifestsCount} volumes`
-                  : imageCount > 0
-                  ? `${imageCount} ${imageCount === 1 ? 'image' : 'images'}`
-                  : ''}
-              </p>
-            </Space>
+                {itemUrl && !audio && !video && (
+                  <Space
+                    as="span"
+                    h={{
+                      size: 'm',
+                      properties: ['margin-right'],
+                    }}
+                  >
+                    <ButtonSolidLink
+                      icon="eye"
+                      text="View"
+                      trackingEvent={{
+                        category: 'WorkDetails',
+                        action: 'follow view link',
+                        label: itemUrl.href.query.workId,
+                      }}
+                      link={{ ...itemUrl }}
+                    />
+                  </Space>
+                )}
+
+                {showDownloadOptions && (
+                  <Download
+                    ariaControlsId="itemDownloads"
+                    workId={work.id}
+                    downloadOptions={downloadOptions}
+                  />
+                )}
+              </div>
+              {!(downloadOptions.length > 0) &&
+                sierraIdFromManifestUrl &&
+                childManifestsCount === 0 && (
+                  <NextLink
+                    {...downloadUrl({
+                      workId: work.id,
+                      sierraId: sierraIdFromManifestUrl,
+                    })}
+                  >
+                    <a>Download options</a>
+                  </NextLink>
+                )}
+              {(childManifestsCount > 0 || imageCount > 0) && (
+                <Space
+                  v={{
+                    size: 'm',
+                    properties: ['margin-top'],
+                  }}
+                >
+                  <p
+                    className={classNames({
+                      'no-margin': true,
+                      [font('lr', 6)]: true,
+                    })}
+                  >
+                    Contains:{' '}
+                    {childManifestsCount > 0
+                      ? `${childManifestsCount} volumes`
+                      : imageCount > 0
+                      ? `${imageCount} ${imageCount === 1 ? 'image' : 'images'}`
+                      : ''}
+                  </p>
+                </Space>
+              )}
+            </>
```

Comment:
With all these pieces of logic, could we write a method that spits out the different know states and use those to decide what to render?

e.g.

```ts
type ItemLinkState = 'showViewer' | `showLibraryLink` | 'blah';
function getItemLinkState(stuff): ItemLinkState {
  //...
}
```